engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/jobrunner/livedata/LiveDataCollectingListenerStorage.scala

```diff
@@ -0,0 +1,94 @@
+package pl.touk.nussknacker.engine.management.jobrunner.livedata
+
+import io.circe.Json
+import pl.touk.nussknacker.engine.api._
+import pl.touk.nussknacker.engine.api.deployment.LiveDataPreviewSupported.LiveDataPreview
+import pl.touk.nussknacker.engine.testmode.TestProcess._
+
+import java.time.Instant
+import java.util.concurrent.{ConcurrentHashMap, ConcurrentLinkedQueue}
+import java.util.concurrent.atomic.AtomicLong
+import scala.jdk.CollectionConverters._
+
+private[livedata] class LiveDataCollectingListenerStorage(
+    maxNumberOfSamples: Int,
+    frequencyWindowInSeconds: Int,
+) {
+
+  private type K = String
+  private type V = TestResults[Json]
+
+  private val results                             = new ConcurrentHashMap[K, V]()
+  private val orderOfResults                      = new ConcurrentLinkedQueue[K]()
+  private val transitionsByEpochSecond            = new ConcurrentHashMap[Long, ConcurrentLinkedQueue[NodeTransition]]()
```

Comment:
Maybe we could just use a `MultiMap` to make it simpler? Or a Map `NodeTransition` ->`InstantRateMeter`?